### PR TITLE
Test m1 runners

### PR DESCRIPTION
**Pull request by samth**

### CODE REVIEW

File: pkgs/racket-benchmarks/tests/racket/benchmarks/treelist/bm.rkt [hunk 1]
```diff
@@ -0,0 +1,298 @@
+#lang racket/base
```
Comment (sorawee) [inline pkgs/racket-benchmarks/tests/racket/benchmarks/treelist/bm.rkt:1]: Accidentally included?

Comment (samth) [reply pkgs/racket-benchmarks/tests/racket/benchmarks/treelist/bm.rkt:1]: Yeah as you can see the branch includes #4913

**Comment by samth** [comment]
Just as a note, this PR took about 13 minutes for the Mac test, whereas #4913 took 28 minutes. So it looks like the M1 runners are quite fast.

**Comment by samth** [comment]
Note that this generates a bunch of (new?) warnings about old C code in zlib:

```

cd racket/src/build/cs/c/ChezScheme/pb/zlib && /Applications/Xcode_15.0.1.app/Contents/Developer/usr/bin/make static
gcc -g -O2 -Wall -DOS_X -DOS_X -DHAVE_HIDDEN -I. -c -o example.o test/example.c
test/example.c:88:6: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]
void test_compress(compr, comprLen, uncompr, uncomprLen)
     ^
```

**Comment by sorawee** [comment]
Should this be reverted? It generates various errors:

https://github.com/racket/racket/actions/runs/7746169161/job/21123775895#step:6:278

**Comment by samth** [comment]
Maybe CGC builds on aarch64 need `--disable-future` explicitly? Probably @mflatt knows.

**Comment by sorawee** [comment]
Also, this PR _switched_ from macos-latest to macos-14 (and not simply adding macos-14). Is this intentional? That is, do we want to drop testing Mac with Intel?

**Comment by mflatt** [comment]
An explicit `--disbale-futures` isn't needed, but the CI rule uses an explicit `--enable-futures`.

**Pull request merged**